src/dclass.d

```diff
@@ -1156,6 +1156,9 @@ public:
                 structsize = Target.ptrsize * 2; // allow room for __vptr and __monitor
         }
 
+        // Add vptr's for any interfaces implemented by this class
+        structsize += setBaseInterfaceOffsets(structsize);
+
```

Comment:
This change affects extern (D) class layout, not only for `extern (C++)`. Test case is:

```d
extern (D) //extern (C++)
{
    class Base
    {
        ~this() {}
        size_t x = 4;
    }

    interface Interface
    {
        int Method();
    }

    class Derived : Base, Interface
    {
        size_t y = 5;
        int Method() { return 3; }
    }
}

void main()
{
    Derived d;                  // dmd 2.069             master
    pragma(msg, d.x.offsetof);  // (void*).sizeof * 2
    pragma(msg, d.y.offsetof);  // (void*).sizeof * 3 -> 4
}
```

Is this intentional? If so, we should update [ABI page](http://dlang.org/spec/abi).